### PR TITLE
[irods/irods#6745] Updated config, documentation, and added pre-commit hook template

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

File: README.md
Comment (alanking) [inline README.md]: Why do we have the contents of .clang-format reproduced in the README?

Comment (korydraughn) [reply README.md]: That is just how it was originally. Can remove if people feel it is unnecessary/wrong.

Comment (trel) [reply README.md]: yeah, let's just link to the real file instead of duplicating the content.  no lag / drift.

Comment (korydraughn) [reply README.md]: Sure thing.

File: README.md
Comment (trel) [inline README.md]: update this to reflect that the 'options above' are no longer above.

File: README.md
Comment (korydraughn) [reply README.md]: Removed it.

I left off the link to the actual file since it's right there.

File: README.md
Comment (korydraughn) [reply README.md]: Nice catch

Comment (korydraughn) [reply README.md]: Done.

**Comment by korydraughn** [comment]
Are there any other changes before I add the pounds?

**Review by alanking — APPROVED** [review]
Looks good to me

**Comment by korydraughn** [comment]
Cool. Waiting on approval from @trel

**Comment by korydraughn** [comment]
Oh, I forgot to add the list of repos that use this configuration.

File: README.md
Comment (trel) [inline README.md:37]: possibly combine these lines...

```
cp /path/to/irods_clang_format_configuration/pre-commit.template ./.git/hooks/pre-commit
```

Comment (trel) [reply README.md:37]: oh, then the chmod line would have to change too...  hmmm, yeah, either update the chmod too - or leave all as is.

Comment (korydraughn) [reply README.md:37]: There's a chmod that follows.

**Review by trel — APPROVED** [review]

**Comment by korydraughn** [comment]
Made some small tweaks to the README.

Please review and let me know if the pounds should be added.

**Review by alanking — APPROVED** [review]

**Comment by korydraughn** [comment]
Changed one word in the README (i.e. `script` to `hook` because `hook` was being used to refer to the pre-commit hook everywhere else).

Added #

**Pull request merged**
